crates/uv-resolver/src/lock.rs

```diff
@@ -763,10 +773,18 @@ impl std::fmt::Display for DistributionId {
     }
 }
 
+/// NOTE: Care should be taken when adding variants to this enum. Namely, new
+/// variants should be added without changing the relative ordering of other
+/// variants. Otherwise, this could cause the lock file to have a different
+/// canonical ordering of distributions.
 #[derive(Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
-struct Source {
-    kind: SourceKind,
-    url: Url,
+enum Source {
+    Registry(Url),
+    Git(Url, GitSource),
+    Direct(Url, DirectSource),
+    Path(PathWire),
```

Comment:
Hmmm I wonder if you could use `PathBuf` directly here and then just use `serialize_with`/`deserialize_with` functions: https://serde.rs/field-attrs.html#deserialize_with. That would let you skip defining a new intermediate `PathWire` type.

(Feel free to get this merged and experiment with it in a follow-up if you want. I'm fine with this as-is.)

Reply:
Good idea, this removes a lot of boilerplate